src/parser/ast.c

```diff
@@ -509,6 +509,23 @@ bool AST_Projects(const AST *ast) {
   return (ast->returnNode || ast->withNode);
 }
 
+bool AST_Empty(const AST *ast) {
+return (!(ast->matchNode != NULL |
+          ast->createNode != NULL |
+          ast->callNode != NULL |
+          ast->returnNode != NULL |
+          ast->mergeNode != NULL |
+          ast->deleteNode != NULL |
+          ast->setNode != NULL |
+          ast->whereNode != NULL |
+          ast->skipNode != NULL |
+          ast->orderNode != NULL |
+          ast->unwindNode != NULL |
+          ast->limitNode != NULL |
+          ast->withNode != NULL |
```

Comment:
These comparisons should be made with `||` (boolean OR) instead of `|` (bitwise OR).

Since the `!=` reduces each statement to 0 or 1, the bitwise version will work, but the boolean is guaranteed to break after the first non-null value.

Reply:
👍